frontend/src/logic/providers/metadata-provider.model.ts

```diff
@@ -0,0 +1,17 @@
+import { VerboseDate } from "../../common-models/month-info.model";
+import { MonthInfoLogic } from "../schedule-logic/month-info.logic";
+
+export abstract class MetadataProvider {
```

Comment:
Zamieniłem interfejsy których używaliśmy na klasy abstrakcyjne, ponieważ to pozwalało podać domyślną implementację do niektórych funkcji. 
Ogólnie, to mam taką proprozycję, żeby używać interfejsów do modeli:
opcje, dane z serwera, pola w stanie, etc.

Natomiast ze wszystkim co ma jakąś logikę używać klas abstrakcyjnych.

Reply:
Rozumiem to w przypadku abstrakcyjnej klasy BaseSectionLogic, bo tam faktycznie domyślna implementacja robic coś sensownego. Jednak czy było to konieczne w kontekście Providerów, skoro tam domyślne implementacje zwracają puste tablice?

Reply:
W sumie tu też można by było dodać domyślne implementacje. Bo `changeFrozenShifts` zawsze będzie działać w podobny sposób. Podobnie też w innych sekcjach, kiedy w jakiś sposób operujemy na danych.
PS. domyślna implementacja `verboseDates`, to hack, który wydał mi się być bardziej sensownym, niż dodanie analogicznej funkcji na poziomie parsera. Ale to potrzeba będzie zmienić na abstrakcyjne property, bo teraz jest trochę dziwnie